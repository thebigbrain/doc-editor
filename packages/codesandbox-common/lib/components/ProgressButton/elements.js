"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Loader = exports.RelativeButton = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Button = require("../Button");

var _theme = _interopRequireDefault(require("../../theme"));

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  top: 50%;\n  left: 50%;\n  transform: translate(-50%, -50%);\n  ", " animation-delay: 0.5s;\n\n  &:before {\n    content: ' ';\n    position: absolute;\n    left: -12px;\n    ", ";\n    animation-delay: 0s;\n  }\n\n  &:after {\n    content: ' ';\n    position: absolute;\n    left: 12px;\n    ", ";\n    animation-delay: 1s;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 8px;\n  height: 8px;\n  border-radius: 50%;\n  background: ", ";\n  opacity: 0.7;\n  animation: ", " 1s infinite linear alternate;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: relative;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  0%   { background-color:  ", "; }\n  50%, 100% { background-color: ", "; }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var loaderAnimation = (0, _styledComponents.keyframes)(_templateObject(), _theme["default"].secondary(), _theme["default"].secondary.lighten(0.5)());
var RelativeButton = (0, _styledComponents["default"])(_Button.Button)(_templateObject2());
exports.RelativeButton = RelativeButton;
var circle = (0, _styledComponents.css)(_templateObject3(), _theme["default"].secondary(), loaderAnimation);

var Loader = _styledComponents["default"].div(_templateObject4(), circle, circle, circle);

exports.Loader = Loader;