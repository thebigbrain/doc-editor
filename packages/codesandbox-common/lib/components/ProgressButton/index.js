"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _elements = require("./elements");

function ProgressButton(_ref) {
  var _ref$loading = _ref.loading,
      loading = _ref$loading === void 0 ? false : _ref$loading,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["loading", "disabled"]);
  return _react["default"].createElement(_elements.RelativeButton, (0, _extends2["default"])({
    disabled: disabled || loading
  }, props), props.children, loading && _react["default"].createElement(_elements.Loader, null));
}

var _default = ProgressButton;
exports["default"] = _default;