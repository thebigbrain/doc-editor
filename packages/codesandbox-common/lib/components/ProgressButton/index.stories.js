"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _ = _interopRequireDefault(require("."));

var stories = (0, _react2.storiesOf)('components/ProgressButton', module);
stories.add('Basic ProgressButton', function () {
  return _react["default"].createElement(_["default"], null, "Click Me");
}).add('ProgressButton disabled', function () {
  return _react["default"].createElement(_["default"], {
    disabled: true
  }, "Click Me");
}).add('ProgressButton loading', function () {
  return _react["default"].createElement(_["default"], {
    loading: true
  }, "Click Me");
});