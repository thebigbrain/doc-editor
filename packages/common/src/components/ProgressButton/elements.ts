import styled, { css, keyframes } from 'styled-components';
import { Button } from '../Button';
import theme from '../../theme';

const loaderAnimation = keyframes`
  0%   { background-color:  ${theme.secondary()}; }
  50%, 100% { background-color: ${theme.secondary.lighten(0.5)()}; }
`;

export const RelativeButton = styled(Button)`
  position: relative;
`;

const circle = css`
  width: 8px;
  height: 8px;
  border-radius: 50%;
  background: ${theme.secondary()};
  opacity: 0.7;
  animation: ${loaderAnimation} 1s infinite linear alternate;
`;

export const Loader = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  ${circle} animation-delay: 0.5s;

  &:before {
    content: ' ';
    position: absolute;
    left: -12px;
    ${circle};
    animation-delay: 0s;
  }

  &:after {
    content: ' ';
    position: absolute;
    left: 12px;
    ${circle};
    animation-delay: 1s;
  }
`;
