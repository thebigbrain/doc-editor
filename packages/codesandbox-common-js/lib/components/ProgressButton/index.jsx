import React from 'react';
import { Loader, RelativeButton } from './elements';
function ProgressButton({ loading = false, disabled = false, ...props }) {
    return (<RelativeButton disabled={disabled || loading} {...props}>
      {props.children}
      {loading && <Loader />}
    </RelativeButton>);
}
export default ProgressButton;
