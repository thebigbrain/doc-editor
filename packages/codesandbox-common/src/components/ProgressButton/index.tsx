import React from 'react';
import { Props as ButtonProps } from '../Button';
import { Loader, RelativeButton } from './elements';

type Props = ButtonProps & {
  loading?: boolean;
};

function ProgressButton({
  loading = false,
  disabled = false,
  ...props
}: Props) {
  return (
    <RelativeButton disabled={disabled || loading} {...props}>
      {props.children}
      {loading && <Loader />}
    </RelativeButton>
  );
}

export default ProgressButton;
