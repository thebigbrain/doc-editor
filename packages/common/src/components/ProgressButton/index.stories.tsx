import React from 'react';
import { storiesOf } from '@storybook/react';
import ProgressButton from '.';

const stories = storiesOf('components/ProgressButton', module);

stories
  .add('Basic ProgressButton', () => <ProgressButton>Click Me</ProgressButton>)
  .add('ProgressButton disabled', () => (
    <ProgressButton disabled>Click Me</ProgressButton>
  ))
  .add('ProgressButton loading', () => (
    <ProgressButton loading>Click Me</ProgressButton>
  ));
