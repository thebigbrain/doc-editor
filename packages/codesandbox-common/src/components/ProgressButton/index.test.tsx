import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import ProgressButton from '.';

describe('<ProgressButton /> rendering', () => {
  it('basic', () => {
    const wrapper = mountWithTheme(<ProgressButton>Click Me</ProgressButton>);
    expect(wrapper).toMatchSnapshot();
  });
  it('disabled', () => {
    const wrapper = mountWithTheme(
      <ProgressButton disabled>Click Me</ProgressButton>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('loading', () => {
    const wrapper = mountWithTheme(
      <ProgressButton loading>Click Me</ProgressButton>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
