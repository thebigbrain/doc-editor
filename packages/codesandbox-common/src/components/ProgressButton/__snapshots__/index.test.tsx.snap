// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressButton /> rendering basic 1`] = `ReactWrapper {}`;

exports[`<ProgressButton /> rendering disabled 1`] = `ReactWrapper {}`;

exports[`<ProgressButton /> rendering loading 1`] = `ReactWrapper {}`;
